fecfile_validate_js/tests/index.test.ts

```diff
@@ -0,0 +1,200 @@
+import { assertEquals } from 'https://deno.land/std@0.133.0/testing/asserts.ts';
+import { validate } from '../dist/index.js';
+import { schema as f3xSchema } from '../dist/F3X.js';
+
+const perfectForm_F3X: any = {
+  form_type: 'F3XA',
+  filer_committee_id_number: 'C00123456',
+  committee_name: 'Foes of Chris',
+  change_of_address: false,
+  street_1: '123 main street',
+  street_2: '',
+  city: 'Best Town',
+  state: 'DC',
+  zip: '20000',
+  report_code: '',
+  election_code: '',
+  date_of_election: '20021101',
+  state_of_election: 'DC',
+  coverage_from_date: '20000101',
+  coverage_through_date: '20000201',
+  qualified_committee: true,
+  treasurer_last_name: 'Doe',
+  treasurer_first_name: 'J',
+  treasurer_middle_name: 'X',
+  treasurer_prefix: 'Dr',
+  treasurer_suffix: 'PhD',
+  date_signed: '20040729',
+  L6b_cash_on_hand_beginning_period: 1,
+};
+
+Deno.test({
+  name: 'it should pass with perfect data',
+  fn: () => {
+    const result = validate(f3xSchema, perfectForm_F3X);
+    assertEquals(result, []);
+  },
+});
+
+Deno.test({
+  name: 'it should fail without form_type',
+  fn: () => {
+    const thisData = { ...perfectForm_F3X };
+    delete thisData.form_type;
+    const result = validate(f3xSchema, thisData);
+    assertEquals(result[0].keyword, 'required');
+    assertEquals(result[0].params.missingProperty, 'form_type');
+  },
+});
+
+Deno.test({
+  name: 'committee_name should allow accents',
+  fn: () => {
+    const thisData = { ...perfectForm_F3X, ...{ committee_name: 'Éàñ!@#$%^&*()_+-=[]\\{}|;,./<>?' } };
+    const result = validate(f3xSchema, thisData);
+    assertEquals(result, []);
+  },
+});
+
+Deno.test({
+  name: "filer_committee_id_number should not be ''",
+  fn: () => {
+    const thisData = { ...perfectForm_F3X, ...{ filer_committee_id_number: '' } };
+    // TODO: the next line is the goal. Ajv sees '' as a value and we want it to be evaluated as null
```

Comment:
why is this the goal? `''` is different from null.  If we configure our schema such that a field is required and the min length is 0, we have approved `''` as a valid entry. I don't believe that we _should_ be allowing `''` but I think that's a matter of setting the `minLength` and or the `pattern` properly.  My reasoning for that is that length is a string property and not a matter of defined or undefined.  `minLenght` and `pattern` are ways that we define string rules.  `required` is how is define whether or not a field must be defined.